skeleton/default.nix

```diff
@@ -0,0 +1,60 @@
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+# Import IOHK common nix lib
+, iohkLib ? import ./nix/iohk-common.nix { inherit system crossSystem config; }
+# Use nixpkgs pin from iohkLib
+, pkgs ? iohkLib.pkgs
+}:
+
+with import ./nix/util.nix { inherit pkgs; };
+
+let
+  haskell = iohkLib.nix-tools.haskell { inherit pkgs; };
+  src = iohkLib.cleanSourceHaskell ./.;
+
+  # Example of using a package from iohk-nix
+  # TODO: Declare packages required by the build.
+  inherit (iohkLib.rust-packages.pkgs) jormungandr;
+
+  # Import the Haskell package set.
+  haskellPackages = import ./nix/default.nix {
+    inherit pkgs haskell src;
+    # Pass in any extra programs necessary for the build as function arguments.
+    # TODO: Declare packages required by the build
```

Comment:
Maybe clarify that `jormungandr` and `cowsay` probably need to be removed :)

Reply:
We've been using `commonLib.nix-tools.default-nix ./pkgs.nix args`. Is that not powerful enough for most projects? Should we rewrite all the other projects to stop using that?

I've also been setting that in `nix/nix-tools.nix` so it can be imported without importing all of `default.nix`.

Reply:
I don't like `commonLib.nix-tools.default-nix` because it hides too many details from the user. If the project `default.nix` just has a plain attrset, then users can see exactly what attributes they will get. There is a little more code, but things become more obvious. The `collectComponents` function can be used to traverse the Haskell.nix package set and pull out all tests or benchmarks, etc. (I should move this to Haskell.nix and document it).